packages/frontend/src/page/components/wallet-position/recover-collateral/index.tsx

```diff
@@ -0,0 +1,148 @@
+    kpiTokenCollateralBalances?: Amount<Token>[];
+    redeemableRewards?: Amount<Token>[] | null;
+}
+
+export const RecoverCollateral = ({
+    t,
+    onTx,
+    kpiToken,
+    collaterals,
+    kpiTokenCollateralBalances,
+}: RecoverCollateralProps) => {
+    const { address } = useAccount();
+    const { chain } = useNetwork();
+    const publicClient = usePublicClient();
+
+    const [loadingRecover, setLoadingRecover] = useState(false);
+    const [collateralToRecover, setCollateralToRecover] =
+        useState<SelectOption<Address> | null>(null);
+
+    const { config: recoverConfig } = usePrepareContractWrite({
+        chainId: chain?.id,
+        address: kpiToken.address as Address,
+        abi: KPI_TOKEN_ABI,
+        functionName: "recoverERC20",
+        args:
+            !!address && !!collateralToRecover
+                ? [collateralToRecover.value, address]
+                : undefined,
+        enabled: !!chain?.id && !!address && !!collateralToRecover,
+    });
+    const { writeAsync: recoverAsync } = useContractWrite(recoverConfig);
+
+    const collateralOptions: OptionForCollateral[] = useMemo(() => {
+        if (!collaterals || !kpiTokenCollateralBalances) return [];
+        return getRecoverableRewards(
+            collaterals,
+            kpiTokenCollateralBalances,
+            kpiToken.expired
+        ).map((collateral) => ({
+            label: `${formatUnits(collateral.raw, 18)} ${
+                collateral.currency.symbol
+            }`,
+            amount: collateral,
+            value: collateral.currency.address,
+        }));
+    }, [collaterals, kpiTokenCollateralBalances, kpiToken.expired]);
+
+    const handleCollateralRecoverClick = useCallback(async () => {
+        if (!recoverAsync || !address || !collateralToRecover) return;
+        setLoadingRecover(true);
+        try {
+            const tx = await recoverAsync();
+            const receipt = await publicClient.waitForTransactionReceipt({
+                hash: tx.hash,
+            });
+            onTx({
+                type: TxType.KPI_TOKEN_COLLATERAL_RECOVER,
+                from: receipt.from,
+                hash: tx.hash,
+                payload: {
+                    receiver: address,
+                    token: collateralToRecover?.value,
+                },
+                receipt: {
+                    from: receipt.from,
+                    transactionIndex: receipt.transactionIndex,
+                    blockHash: receipt.blockHash,
+                    transactionHash: receipt.transactionHash,
+                    to: receipt.to || zeroAddress,
+                    contractAddress: receipt.contractAddress || zeroAddress,
+                    blockNumber: Number(receipt.blockNumber),
+                    status: receipt.status === "success" ? 1 : 0,
+                },
+                timestamp: unixTimestamp(new Date()),
+            });
+            setCollateralToRecover(null);
+        } catch (error) {
```

Comment:
Let's maybe add a console log here. In case an error happens, it will be easier to debug. Something like:

```
console.error(`could not recover collateral ${collateral.currency.address}`, error)
```